docs/cluster-guide.md

```diff
@@ -0,0 +1,108 @@
+
+
+
+## Preparing the environment
+
+1. Create a folder named 'Cluster'. Within it, create three subfolders: 'Node1', 'Node2', and 'Node3'.
+2. In each node folder, add a configuration file (either .yml or .txt).
+3. Within each node folder, create the following subfolders:
+    - 'Certificates' to store the certificate and private key. 
+    - 'Data' for all data files.
+    - 'Index' for node indexes.
+    - 'Logs' for log files.
+
+
+
+## Generating certificates
+
+1. Create a folder named 'Generate_Certificate'.
+2. Download the latest version of the certificate generator from [EventStore 'es-gencert-cli'-releases](https://github.com/EventStore/es-gencert-cli/releases) and unzip it into the 'Generate_Certificate' folder.
+3. In a terminal, change to the `es-gencert-cli` directory: 
+
+
+Run this command to generate the root certificate and root private key on Linux: 
+`./es-gencert-cli create-ca -out [Generate_Certificate Path]/ca`
+
+On Windows:
+
+`.\es-gencert-cli.exe create-ca -out [Generate_Certificate Path]\ca`
+
+
+
+4. In the 'Generate_Certificate' directory, run the following commands one at a time for each node (change the path and node number as needed) to generate the certificate and private keys for each node. Use your actual DNS names and recommended IP address configurations to align with production practices. For further details, check out our [cluster with DNS guide](https://developers.eventstore.com/server/v23.10/cluster.html#cluster-with-dns).
+
+For example, if the certificate generator version is 1.2.1 and we’re generating the certificate and private keys for 'Node1' on Linux:
+
+`./es-gencert-cli create-node -ca-certificate /path/to/folder/Generate_Certificate/es-gencert-cli_1.2.1_Linux-x86_64/ca/ca.crt -ca-key` 
+`/path/to/folder/Generate_Certificate/es-gencert-cli_1.2.1_Linux-x86_64/ca/ca.key -out`
+`/path/to/folder/Cluster/Node1/certificates -dns-names your.node1.dns.com`
+
+ The command line will create a folder named 'Certificates' containing the certificate and private key for each node in the directory where we created the folder for Node1 in the folder cluster. 
+
+:::
+Each command will automatically generate the security certificate and the private key for each node in their respective certificates file:
+
+Linux: `/path/to/folder/Cluster/Node1/certificates` 
+Windows: `C:\Path\To\Folder\Cluster\Node1\certificates`
+
+5.  Include CA certificate and key paths in each node's configuration file. An example of a complete configuration file on Linux:
+
+``` 
+# Paths
+Db: /path/to/folder/Cluster/Node1/Data
+Index: /path/to/folder/Cluster/Node1/Index
+Log: /path/to/folder/Cluster/Node1/Log
+
+# Network configuration
+IntIp: your.node1.internal.dns.com
+ExtIp: your.node1.external.dns.com
+HttpPort: 21131
+IntTcpPort: 11121
+ExtTcpPort: 11131
```

Comment:
External TCP options should be removed since they are not supported as of 24.2. See https://github.com/EventStore/EventStore/pull/4166/files#diff-66f272e2a92c2af962ff164473dad92d1aff2ac2a7f616f1bccf8ba1f192aeac